core-ui/src/main/java/io/github/droidkaigi/confsched2022/ui/Result.kt

```diff
@@ -13,10 +13,41 @@ sealed interface Result<out T> {
 }
 
 fun <T> Flow<T>.asResult(): Flow<Result<T>> {
-    return this
-        .map<T, Result<T>> {
-            Result.Success(it)
+    return this.map<T, Result<T>> {
+        Result.Success(it)
+    }.onStart { emit(Result.Loading) }.catch { emit(Result.Error(it)) }
+}
+
+sealed interface LoadState<out T> {
```

Comment:
I can understand what you want to do. But this model has the same data.
So probably we may be confused about which to use. So how about renaming it to UiLoadState.

Reply:
Sure I can rename this!

Reply:
Done 🎇